vector/src/main/java/im/vector/app/features/spaces/explore/SpaceDirectoryController.kt

```diff
@@ -110,6 +110,7 @@ class SpaceDirectoryController @Inject constructor(
                     ?.filter {
                         it.parentRoomId == (data.hierarchyStack.lastOrNull() ?: data.spaceId)
                     }
+                    ?.filterNot { it.isUpgradedRoom(data) }
```

Comment:
it will filter out rooms with `UPGRADED_ROOM_NOT_JOINED` VersioningState, is it intentional? If room was upgraded, but user haven't joined upgraded room yet, what user will see? We change state from `NONE` to `UPGRADED_ROOM_NOT_JOINED` on TombStone event and so it could happen while user is on a background and doesn't interact with the app.

Reply:
This is intentional yeah, this only applies to the old version of upgraded rooms, so them not being active there's no use having them shown in Explore Rooms

Reply:
Just to clarify, we should filter out if the room is an old version of a room that is also in the space (we want to avoid apparant duplicates in the list).

Reply:
So, when room is upgraded, everyone in the room gets tombstone event and state is changed to `UPGRADED_ROOM_NOT_JOINED` until they will join new room. `isUpgradedRoom` will return true and so such users won't see room in a list. Shouldn't we hide it only after users will join new Room?

Reply:
But you don't keep it, you filter out any room with `UPGRADED_ROOM_NOT_JOINED`, so it's different from web, no?

Reply:
I mean keep in the sense that it's not deleted from the space in the db, it's just filtered out in the UI